buildSrc/src/main/groovy/Docker.groovy

```diff
@@ -337,35 +364,34 @@ class Docker {
                 }
             }
 
-            // Handle copying failure. This is now distinct from the "actual" Sync task that depends directly
-            // on the CombinedDockerRunTask.
-            def syncAfterFail = project.tasks.register("${taskName}SyncAfterFail", Sync) { sync ->
+            // Specify that makeImage is finalized by buildAndRun - that is, in this configuration buildAndRun
+            // must run after makeImage finishes
```

Comment:
Your comment is a bit at odds with the typical gradle terminology. I'm digging in now; I don't _know_ if finalizedBy is the technically correct solution here (it may very well be...).

https://docs.gradle.org/current/userguide/more_about_tasks.html#sec:finalizer_tasks
> Finalizer tasks are useful in situations where the build creates a resource that has to be cleaned up regardless of the build failing or succeeding. 

https://docs.gradle.org/current/userguide/more_about_tasks.html#sec:ordering_tasks
> When you use the “must run after” ordering rule you specify that taskB must always run after taskA, whenever both taskA and taskB will be run. This is expressed as taskB.mustRunAfter(taskA).

Reply:
`mustRunAfter`/`shouldRunAfter` only sets an ordering requirement, but does not mandate that the earlier or later task run at all, they can run or not run independently.

On the other hand, `b.dependsOn(a)` means that if `b` is requested to run, `a` must run first and must be successful, whereas `a.finalizedBy(b)` means that if `a` will run, `b` must run _afterwards_ (regardless of whether or not `a` is successful).